### DISCUSSION TITLE
What is the page limit of a Docusaurus site?

### DISCUSSION THREAD

**Discussion by i-adarsh** [facebook/docusaurus#10557] [Q&A]
I've around 20k docs (pages) in my site which I'm trying to deploy, but the server is not getting started and don't even build. I've tested for 5 - 6 docs then it was working fine, but as soon as I have added all my files it didn't start and build.  It is getting failed with the below error [1].

Any suggestion on how I can build 20k pages document site with docusaurus?

>Just FYI: I've setup the node memory limit to 90GB

```
export NODE_OPTIONS="--max-old-space-size=92160"
```

[1]
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by slorber** [comment]
There is no specific page limit and we have seen complex sites with 20k+ pages already.

I doubt allocating 90Gb of memory to your node process is a good idea, unless your system actually have such amount of memory. This might prevent the JS runtime to GC temporary objects as it thinks it has available memory while everything is swapping. Maybe try with a more reasonable number like 8GB, 12GB, 16GB?

You might find this issue useful: https://github.com/facebook/docusaurus/issues/4765

Maybe you could use the `DOCUSAURUS_PERF_LOGGER=true` env variable and tell us at which step Docusaurus crashes? (show me the full logs, I'll figure it out)

https://github.com/facebook/docusaurus/issues/4765#issuecomment-2066227919

Creating a runnable OSS repro could help me troubleshoot it.

---

These upcoming `faster` options coming in v3.6 could maybe help if you are ok with using a canary version of Docusaurus until we release those: https://github.com/facebook/docusaurus/issues/10556

**Comment by i-adarsh** [comment]
Thank you so much for the response 🙏 

I've allocated 16GB this time and the exported the env variable `export DOCUSAURUS_PERF_LOGGER=true`, but still facing the same issue. The build got stuck in between for ~1h.

![3xgWBUxLQvrQtyX](https://github.com/user-attachments/assets/b09dd188-fbd2-4ab5-8c4e-9c604c260afc)

**Reply by slorber** [reply]
Looks like it's getting stuck at the bundling phase. 

I suspect it might be Webpack persistent caching system. It's possible that disabling it would fix the issue, but we don't have any config option for that atm. Try removing the `cache` property locally in this file and tell us please: `/node_modules/@docusaurus/core/lib/webpack/base.js`

It's possible that the upcoming Rspack support solves it. (see https://github.com/facebook/docusaurus/issues/10556)

**Comment by i-adarsh** [comment]
Thank you so much for the response, I've followed the suggestion and its still getting stuck in between:

`/node_modules/@docusaurus/core/lib/webpack/base.js`, please LMK if I'm commenting the exact lines, or else I've to remove or add any other lines?

<img width="894" alt="Screenshot 2024-10-11 at 8 04 45 PM" src="https://github.com/user-attachments/assets/81bd0053-5a21-4f7c-8d63-7afc772e85a1">

`npx docusaurus build`

<img width="1089" alt="Screenshot 2024-10-11 at 8 05 38 PM" src="https://github.com/user-attachments/assets/74ebc56d-4f02-4116-a276-821914cd6cf4">

I'm waiting for the `Docusaurus Faster v3.6`, do we have any ETA for this? 🙏

**Reply by slorber** [reply]
You don't have to wait, use a canary version today: `0.0.0-6101`

See for example this upgrade PR: https://github.com/facebook/react-native-website/pull/4268
